core/cas-server-core-rest/src/main/java/org/apereo/cas/rest/factory/RestHttpRequestCredentialFactory.java

```diff
@@ -24,6 +25,16 @@
      * @return Credential instance
      */
     List<Credential> fromRequestBody(MultiValueMap<String, String> requestBody);
+    
+    /**
+     * 
+     * @param request object containing headers to extract the X509Certificate(s) from
+     * @param requestBody to optionally construct Credential from - if x509insecure is set to true
```

Comment:
Based on my note below, you should try to remove this method altogether, and simply add the request parameter to the existing method. That way, `if x509insecure is set to true` this goes away.